legal-api/src/legal_api/services/filings/validations/validation.py

```diff
@@ -88,6 +89,26 @@ def validate(business: Business, filing_json: Dict) -> Error:  # pylint: disable
                                                       'path': '/filing/specialResolution'}])
         if err:
             return err
+    elif 'specialResolution' in filing_json['filing'].keys() and legal_type in [Business.LegalTypes.COOP.value]:
+        err = special_resolution_validate(business, filing_json)
+        if err:
+            return err
+
+        either_con_or_alteration_flag = False
+
+        if 'changeOfName' in filing_json['filing'].keys():
+            either_con_or_alteration_flag = True
+            err = con_validate(business, filing_json)
+        if 'alteration' in filing_json['filing'].keys():
+            either_con_or_alteration_flag = True
+            err = alteration_validate(business, filing_json)
+
+        if err:
+            return err
+
+        if not either_con_or_alteration_flag:
+            err = Error(HTTPStatus.BAD_REQUEST, [{'error': babel('Either Change of Name or Alteration is required.'),
```

Comment:
I am not sure about this. @thorwolpert can you confirm this is how it should work for COOP.

Reply:
I believe this is correct.  

The four possible scenarios are as follows.  scenario 1 is basically irrelevant as it doesn't require a filing.
1. address change is made and is instantly saved -> no filing is required
2. only articles changes -> special resolution + alteration
3. CoN + articles changes -> special resolution + CoN + alteration
4. CoN -> special resolution + CoN

I still think some additional validation will need to happen around the rules and memorandum.  For example, key and filename are both required for rules and memorandum doc updates.  If only one of key or filename are provided, we will need to provide a validation error.

Reply:
Added the new validations